llumnix/llumlet/local_migration_scheduler.py

```diff
@@ -39,17 +45,23 @@ def get_migrate_out_requests(self, min_request_len=0, max_request_len=np.inf) ->
                 migrate_out_requests = self._get_first_waiting_and_shortest_running_requests(min_request_len, max_request_len)
         return migrate_out_requests
 
-    # The function is used to retrieve requests on the backend that have already met the expected_steps.
-    # (xinyi): Currently, the function is only used for Prefill-decode disaggregation,
-    # and only selects request that migrates from the prefill instance to the decode instance.
-    def get_required_migration_request(self):
-        running: List[LlumnixRequest] = self.backend_engine.get_running_queue()
+    def get_required_migration_request(self, migration_type: Optional[MigrationType] = None):
         required_migration_requests = []
-        for request in reversed(running):
-            if request.llumnix_status == RequestStatus.RUNNING \
-                and request.inference_type == RequestInferenceType.DECODE \
-                and request.output_len >= request.expected_steps:
-                required_migration_requests.append(request)
+        if migration_type == MigrationType.FAILOVER_MIGRATION:
```

Comment:
put failover migration in else branch, because failover migration is a special case.